### PR TITLE
Add a drop down menu to snTable header

### PR DESCRIPTION
**Pull request by Donya-qlik**
The sn-table dropdown menu contains a menu button that opens the context menu when it is clicked. The menu currently has two options to sort the column cell values ascending/descending:
- Sort ascending
- Sort descending

The menu button is hidden by default, but will be visible as soon as the mouse hovers the head cell or the head cell is on focus or clicked. Clicking anywhere out side the head cell results the context menu to be closed and the menu button returns to the hidden mode.
To see the changes, table feature flag should be toggled: `PS_15585_SN_TABLE_BASIC_FEATURES`

### CODE REVIEW

File: src/table/components/HeadCellMenu.tsx
Comment (Donya-qlik) [reply src/table/components/HeadCellMenu.tsx:36]: I tried two couple of options but because of the on the way issues couldn't make them complete:
- Not using stopPtopagation but instead making a callback that confirms the menu is open so disabling sorting function, but because of some reasons it couldn't update the setState value at the right time.
- Using stopPropagation  but moving the ClickAwayEventListener to very higher level component to TableCell but then one of useRef hooks didn't get the right value and a console error came up that I couldn't resolve it.
Which do you think might make more sense to continue with?

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [inline src/table/components/HeadCellMenu.tsx:21]: do you really need the button component here?

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [inline src/table/components/HeadCellMenu.tsx:33]: I think this can mean trouble. We want a single source of truth whether it is open or not

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [reply src/table/components/HeadCellMenu.tsx:36]: Well, the main problem is that we have a button (open menu) inside a different button (sort). This is bad design. Not sure how we solve that, because it depends on the a11y design for the header... What I would do for now is to try to treat the sort "button" like it has a hole in it, where the menu button is. In the click handler for the sorting, check if the event comes from the menu button, and then early return.

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [inline src/table/components/HeadCellMenu.tsx:41]:
```suggestion
    !(anchorRef.current?.contains(event.target as HTMLElement)) && setOpen(false);
```

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [inline src/table/components/HeadCellMenu.tsx:55]: use the sn-table prefix for consistency `sn-table-head-menu-button`

File: src/table/hooks/use-head-cell-menu.ts
Comment (haxxmaxx) [reply src/table/hooks/use-head-cell-menu.ts:4]: hmm no not really adding any value I would say, just makes it harder to follow. the open states are very much tied to the component

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [inline src/table/components/HeadCellMenu.tsx:80]: `sn-table-head-menu`

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [inline src/table/components/HeadCellMenu.tsx:81]: This is not a navigation menu, should not need the nav tag

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [inline src/table/components/HeadCellMenu.tsx:83]: will need to add proper strings for this eventually

File: src/table/components/TableHeadWrapper.tsx
Comment (haxxmaxx) [inline src/table/components/TableHeadWrapper.tsx:80]: this is not what this component is, more like HeadCellContent

**Review by haxxmaxx — COMMENTED** [review]
I will checkout the branch and see if I can get it to work

File: src/table/components/HeadCellMenu.tsx
Comment (Donya-qlik) [reply src/table/components/HeadCellMenu.tsx:21]: Can't say surely it is. I mostly followed the standard context menu I found in different examples, for me it is more like a button in the context menu that is pressed and the event fires.

File: src/table/components/HeadCellMenu.tsx
Comment (Donya-qlik) [reply src/table/components/HeadCellMenu.tsx:36]: ok I'll remove it but what should I put instead? just the text with the bigger size padding ?

File: src/table/components/HeadCellMenu.tsx
Comment (Donya-qlik) [reply src/table/components/HeadCellMenu.tsx:81]: True, it is not really needed.

File: src/table/components/HeadCellMenu.tsx
Comment (Donya-qlik) [reply src/table/components/HeadCellMenu.tsx:83]: Do you mean the itemTitle text? It is what design team has chosen.

File: src/table/hooks/use-head-cell-menu.ts
Comment (Donya-qlik) [reply src/table/hooks/use-head-cell-menu.ts:4]: Sure, I'll delete it.

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [reply src/table/components/HeadCellMenu.tsx:36]: no you wont actually change the styling. You just treat it as a "hole" in the click handler.
if (event.target === button) return

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [reply src/table/components/HeadCellMenu.tsx:83]: yes but we need to add it to translation, so we call translator.get('somStringName') so it works for all languages

File: src/table/components/HeadCellMenu.tsx
Comment (Donya-qlik) [reply src/table/components/HeadCellMenu.tsx:36]: Aha you mean 'sort' button, sorry I misunderstood first, got it now.

File: src/table/components/HeadCellMenu.tsx
Comment (Donya-qlik) [reply src/table/components/HeadCellMenu.tsx:83]: Should it be added in sense-client or there is a place in sn-table repo for it ?

**Review by haxxmaxx — COMMENTED** [review]
Also this needs to be behind flag

File: src/locale/locales/en.json
Comment (haxxmaxx) [inline src/locale/locales/en.json:55]: remove the dot

File: src/table/utils/handle-click.ts
Comment (haxxmaxx) [inline src/table/utils/handle-click.ts:34]: no need to put it here. My point was to pit it ouside the elements, like we do for `handleKeyDown`

File: src/table/styles.ts
Comment (haxxmaxx) [inline src/table/styles.ts:160]: the headerStyle prop does not exist here, so this one should be static. There are a couple of examples of the syntax for this, e.g. `VisuallyHidden`

File: src/table/components/__tests__/HeadCellMenu.spec.tsx
Comment (Donya-qlik) [inline src/table/components/__tests__/HeadCellMenu.spec.tsx:51]: It's a weird situation, I've already modified the SortItem to MenuItem, but it only passes the test with these strings! 
As I've shown you, the translator doesn't work properly for me.

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [inline src/table/components/HeadCellMenu.tsx:49]: set tabindex to -1 for now, so it is not possible to tab here

File: src/table/components/TableHeadWrapper.tsx
Comment (haxxmaxx) [inline src/table/components/TableHeadWrapper.tsx:63]: this does not need to be a function

File: src/table/components/__tests__/HeadCellMenu.spec.tsx
Comment (haxxmaxx) [inline src/table/components/__tests__/HeadCellMenu.spec.tsx:52]: this can be the same test

File: src/table/styles.ts
Comment (haxxmaxx) [inline src/table/styles.ts:150]: I think this and the one for the menu can be inherited

File: src/table/components/HeadCellMenu.tsx
Comment (haxxmaxx) [inline src/table/components/HeadCellMenu.tsx:51]: I think you want this to be id, otherwise you are breaking the aria-labeledby for the menu

File: src/table/components/TableHeadWrapper.tsx
Comment (haxxmaxx) [inline src/table/components/TableHeadWrapper.tsx:99]: wait, why are you only using `HeadCellContent` when the flag is on? I think it could be there for both cases. Tried it quickly and it looks better when the arrows are next to the text, more like the design. Will save some work when I fix the styling later

Comment (Donya-qlik) [reply src/table/components/TableHeadWrapper.tsx:99]: What do you mean? when the flag is on it shows everything as the design says.  When it is off it looks exactly as same as it is on the old table. shouldn't it be like that  with flag

Comment (Donya-qlik) [reply src/table/components/TableHeadWrapper.tsx:99]: I guess you might forgot to set the flag to `true` when you run it on nebula ?

File: src/table/components/TableHeadWrapper.tsx
Comment (haxxmaxx) [inline src/table/components/TableHeadWrapper.tsx:107]:
```suggestion
              <HeadCellContent>
                <StyledSortLabel
                  headerStyle={headerStyle}
                  active={isCurrentColumnActive}
                  title={!constraints.passive ? column.sortDirection : undefined} // passive: turn off tooltips.
                  direction={column.sortDirection}
                  tabIndex={-1}
                  onMouseDown={(evt: React.MouseEvent) =>
                    handleMouseDownLabelToFocusHeadCell(evt, rootElement, columnIndex)
                  }
                >
                  <CellText>{column.label}</CellText>
                  {isFocusInHead && (
                    <VisuallyHidden data-testid={`VHL-for-col-${columnIndex}`}>
                      {translator.get('SNTable.SortLabel.PressSpaceToSort')}
                    </VisuallyHidden>
                  )}
                </StyledSortLabel>
                {areBasicFeaturesEnabled && <HeadCellMenu headerStyle={headerStyle} translator={translator} />}
              </HeadCellContent>
```

File: src/table/components/TableHeadWrapper.tsx
Comment (haxxmaxx) [reply src/table/components/TableHeadWrapper.tsx:63]: remove this variable, see my suggestion below

File: src/table/components/TableHeadWrapper.tsx
Comment (Donya-qlik) [reply src/table/components/TableHeadWrapper.tsx:107]: AAh, yes off course! I forgot to refactor it when both are with HeadCellContent

File: src/table/components/__tests__/TableHeadWrapper.spec.tsx
Comment (Donya-qlik) [inline src/table/components/__tests__/TableHeadWrapper.spec.tsx:86]: I'm going to skip this for now. I couldn't verify that the button is visible after focusing the head cell element. The focus is a css focus so I don't see any other option to focus on it than what it is now. It used to work before rebasing the branch. By the way, the current scenario works properly in reality on the table.

File: src/table/components/TableHeadWrapper.tsx
Comment (haxxmaxx) [inline src/table/components/TableHeadWrapper.tsx:64]: you can remove this now since the menu is below the header

File: src/table/components/__tests__/TableHeadWrapper.spec.tsx
Comment (haxxmaxx) [reply src/table/components/__tests__/TableHeadWrapper.spec.tsx:86]: ok that is fine

**Review by haxxmaxx — APPROVED** [review]
looks good for now. All the styling will probably be adjusted when there is a pattern in place and we are using the complete sprout. Wait for string approvals before you merge

**Review by tklkan — APPROVED** [review]

**Pull request merged**
